drivers/Aqara/aqara-lock/src/credential_utils.lua

```diff
@@ -0,0 +1,151 @@
+local capabilities = require "st.capabilities"
+local utils = require "st.utils"
+
+local remoteControlStatus = capabilities.remoteControlStatus
+local lockCredentialInfo = capabilities["stse.lockCredentialInfo"]
+
+local credential_utils = {}
+
+local HOST_COUNT = "__host_count"
+
+local function save_data(driver)
+  driver.datastore:save()
+  driver.datastore:commit()
+end
```

Comment:
In general, I think this should happen for you seamlessly without needing to invoke these methods directly. @varzac may have more of an opinion, but the datastore is saved frequently.

Reply:
It's possible for the auth data stored in the datastore to be lost if the hub loses power before the datastore is synced to nonvol. Even though this was anticipated to be exceedingly rare in the field it was determined that we would need to support a mechanism to force the immediate sync of the datastore to nonvol (the `commit`api). So these two calls are necessary to ensure that the auth data can't be lost.